src/main/java/seedu/utility/DataManager.java

```diff
@@ -0,0 +1,88 @@
+package seedu.utility;
+
+import seedu.duke.Parser;
+import seedu.entry.Expense;
+import seedu.entry.Income;
+import seedu.exceptions.InvalidExpenseAmountException;
+import seedu.exceptions.InvalidExpenseDataFormatException;
+import seedu.exceptions.InvalidIncomeAmountException;
+import seedu.exceptions.InvalidIncomeDataFormatException;
+
+import java.io.BufferedWriter;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.DateTimeException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Saves entries that StonksXD is currently tracking into a csv file so that entries can be saved and be easily
+ * manipulated by user.
+ * Also loads all saved entries when a new instance of StonksXD starts.
+ */
+public class DataManager {
+    private final String FILENAME = "./StonksXD_Data.csv";
+
+    public void save(Parser parser, FinancialTracker financialTracker, Ui ui) {
+        try {
+            FileWriter writer = new FileWriter(FILENAME);
+            BufferedWriter buffer = new BufferedWriter(writer);
+            ArrayList<Expense> expenses = financialTracker.getExpenses();
+            ArrayList<Income> incomes = financialTracker.getIncomes();
+            String data;
+            String NEWLINE = "\n";
+
+            // Categories header for the CSV file
+            buffer.write("entry_type, entry_description, amount, category, date");
+            buffer.write(NEWLINE);
+            for (Expense expense : expenses) {
+                data = parser.convertExpenseToData(expense);
+                buffer.write(data);
+                buffer.write(NEWLINE);
+            }
+            for (Income income : incomes) {
+                data = parser.convertIncomeToData(income);
+                buffer.write(data);
+                buffer.write(NEWLINE);
+            }
+
+            buffer.close();
+        } catch (IOException e) {
+            ui.printError(Messages.ERROR_SAVING_ENTRIES);
+        }
+    }
+
+    public void load(Parser parser, FinancialTracker financialTracker, Ui ui) {
```

Comment:
Would it be better to throw exceptions that contain error messages from this method? This might reduce the number of try catch blocks making the code more readable

Reply:
I feel that it is fine currently. I am also not too sure how to implement your method, maybe you can try first?